test/test_torch.py

```diff
@@ -2621,6 +2621,14 @@ def _test_gather(self, cast, test_bounds=True):
             bad_src = torch.randn(*[i - 1 for i in idx_size])
             self.assertRaises(RuntimeError, lambda: torch.gather(bad_src, dim, idx))
 
+            # should throw an error when index dtype is not long
+            with self.assertRaisesRegex(RuntimeError, 'Expected dtype int64 for index'):
+                torch.gather(src, dim, idx.to(torch.int))
```

Comment:
Wouldn't `torch.int32` be more future proof?
```suggestion
                torch.gather(src, dim, idx.to(torch.int32))
```

Reply:
I don't think we'll change this.  Also, this test is just matching the existing scatter test (see scatter_base) -- I'm not claiming it's the best test :).